### PR TITLE
Add support for notification window

### PR DESCRIPTION
**Pull request by aszulak**

### CODE REVIEW

File: runtime/browser/native_app_window.cc [hunk 1]
```diff
@@ -26,9 +27,24 @@ NativeAppWindow::NativeAppWindow() {
 NativeAppWindow::~NativeAppWindow() {
 }
 
-Evas_Object* NativeAppWindow::CreateWindowInternal() {
+Evas_Object* NativeAppWindow::CreateWindowInternal(
```
Comment (myroot) [inline runtime/browser/native_app_window.cc]: You should create a new NativeWindow class that inherited NativeWindow

**Comment by myroot** [comment]
You should make a new window class.
ex) 
class Notification : public NativeWindow {
  ....
}

and Do Not change the NativeWindow Interface

**Comment by tiwanek** [comment]
Squash commits and push force to the same branch in your clone of repository

File: runtime/browser/runtime.cc [hunk 2]
```diff
@@ -29,16 +29,41 @@
...
+  switch (category) {
+    case NativeWindow::WindowCategory::NORMAL: {
+      auto cached = PreloadManager::GetInstance()->GetCachedNativeWindow();
+      window = cached != nullptr ? cached : new NativeAppWindow();
+    }
+    break;
+    case NativeWindow::WindowCategory::ALWAYS_ON_TOP: {
+      window = new NotificationWindow();
+    }
+    break;
```
Comment (myroot) [inline runtime/browser/runtime.cc]: cached window must be deleted, if dose not used

File: runtime/browser/runtime.cc [hunk 3]
```diff
@@ -29,16 +29,45 @@
...
+  NativeWindow* window;
+
+  switch (category) {
+    case NativeWindow::WindowCategory::NORMAL: {
+      auto cached = PreloadManager::GetInstance()->GetCachedNativeWindow();
+      window = cached != nullptr ? cached : new NativeAppWindow();
+    }
+    break;
+    case NativeWindow::WindowCategory::ALWAYS_ON_TOP: {
+      auto cached = PreloadManager::GetInstance()->GetCachedNativeWindow();
```
Comment (myroot) [inline runtime/browser/runtime.cc]: GetCachedNativeWindow() was used release of unique_ptr
So Just call ReleaseCachedNativeWindow(), without check of cached nativewindow

PreloadManager::GetInstance()->ReleaseCachedNativeWindow();
window = new NotificationWindow();

**Comment by myroot** [comment]
LGTM

File: runtime/browser/native_window.cc [hunk 4]
```diff
@@ -216,4 +219,34 @@ void NativeWindow::FullScreen(bool enable) {
       enable ? ELM_WIN_INDICATOR_TRANSPARENT : ELM_WIN_INDICATOR_OPAQUE);
 }
 
+void NativeWindow::ToggleOnTopProperty(bool status) {
```
Comment (myroot) [inline runtime/browser/native_window.cc]: Don't use elm_win_aux_hint 
Just add property into NativeWindow class

File: runtime/browser/web_application.h [hunk 5]
```diff
@@ -90,11 +90,13 @@ class WebApplication : public WebView::EventListener {
...
-  virtual void OnRotaryEvent(
-      WebView* view, RotaryEventType type);
-#endif  // ROTARY_EVENT_FEATURE_SUPPORT
+        WebView* view, SoftKeyboardChangeEventValue softkeyboard_value);
+  #ifdef ROTARY_EVENT_FEATURE_SUPPORT
+    virtual void OnRotaryEvent(
+        WebView* view, RotaryEventType type);
+  #endif  // ROTARY_EVENT_FEATURE_SUPPORT
+
+    void SetupOnTopProperty(common::AppControl* appcontrol);
```
Comment (myroot) [inline runtime/browser/web_application.h]: Don't make as public

File: runtime/browser/native_window.cc [hunk 6]
```diff
@@ -216,4 +219,34 @@ void NativeWindow::FullScreen(bool enable) {
...
+  if (status) {
+    if (auxidLock_ == -1) {
+      auxidLock_ = elm_win_aux_hint_add(window_, "wm.policy.win.above.lock",
```
Comment (myroot) [inline runtime/browser/native_window.cc]: What is this?
Can I refer behaivor of "wm.policy.win.above.lock" ?

If we can control the Always top attribute through this options. 
we need not make notification window class

**Comment by myroot** [comment]
I think, you misunderstand the requirements

Window should be Re-Created in OnAppControl Handler
If always_on_top value was changed

**Pull request closed without merging**